core/lib/makeswift/components/customer-group-slot/customer-group-slot.makeswift.tsx

```diff
@@ -0,0 +1,141 @@
+import { Checkbox, Combobox, Group, List, Slot, Style } from '@makeswift/runtime/controls';
+import { ReactNode } from 'react';
+import useSWR from 'swr';
+import { z } from 'zod';
+
+import { CustomerGroupsSchema, CustomerGroupsType } from '~/app/api/customer/groups/route';
+import { GetCustomerGroupResponse } from '~/client/queries/get-customer';
+import { runtime } from '~/lib/makeswift/runtime';
+
+const CustomerGroupSchema = z.object({
+  customer: z.object({
+    customerGroupId: z.number(),
+  }),
+});
+
+interface Props {
+  className: string;
+  slots?: Array<{ group?: string; slot: ReactNode }>;
+  simulateGroup: boolean;
+  simulatedGroup?: string;
+  noGroupSlot: ReactNode;
+}
+
+async function getAllCustomerGroups(): Promise<CustomerGroupsType> {
+  const response = await fetch('/api/customer/groups');
+
+  if (!response.ok) {
+    throw new Error(`HTTP error! status: ${response.status}`);
+  }
+
+  const data: unknown = await response.json();
+  const groups = CustomerGroupsSchema.parse(data);
+
+  return groups;
+}
+
+async function fetchCustomerGroupData(): Promise<GetCustomerGroupResponse> {
+  const response = await fetch('/api/customer/group');
+
+  if (!response.ok) {
+    throw new Error(`HTTP error! status: ${response.status}`);
+  }
+
+  const data: unknown = await response.json();
+  const group = CustomerGroupSchema.parse(data);
+
+  return group;
+}
+
+function CustomerGroupSlot({
+  className,
+  slots,
+  simulateGroup,
+  simulatedGroup = 'no-group',
```

Comment:
Can we extract this into a constant instead of using a mage string here and [below](https://github.com/bigcommerce/catalyst/pull/1970/files#diff-a30ece14842040e4b499de1748068dbdf68baab6e40c80cee449e227cfc7e27aR133)?